src/renderer/hooks/useMap.ts

```diff
@@ -0,0 +1,47 @@
+// From https://usehooks-ts.com/react-hook/use-map
```

Comment:
Why did you copy a whole copy-on-write-map hook implementation? Do you expect to use more than `set` in the future?

Also, the implementation and types aren't even that good. If we decide to keep it, I'll do a followup PR to bring it in shape.

Reply:
I honestly just wasnt sure what the best way to store this data was

Reply:
Storing it in a map is the right decision IMO, I just don't like the hook. Seems overkill to me if we use it for just the broadcast outputs.

Reply:
I'm willing to change it then. Should I just use a non-hook map?